🏗 Move Edge testing from GH Actions to CircleCI

rsimha opened the pull request:
Picking up from where #35455 and #35594 left off.

**PR Highlights:**
- Move all Edge testing to CircleCI
- Reuse build output from the previous stage
- Clean up remaining bits of CI on GH Actions (rename / refactor workflow config and node script)

estherkim commented on build-system/tasks/runtime-test/runtime-test-base.js:
```diff
@@ -116,82 +117,21 @@ class RuntimeTestConfig {
-        browsers: [argv.headless ? 'EdgeHeadless' : 'Edge'],
-      });
-      return;
+      Object.assign(this, {browsers: ['EdgeCustom'], customLaunchers});
+    } else if (argv.firefox) {
+      Object.assign(this, {browsers: ['FirefoxCustom'], customLaunchers});
+    } else if (argv.ie) {
+      Object.assign(this, {browsers: ['IECustom'], customLaunchers});
+    } else if (argv.safari) {
+      Object.assign(this, {browsers: ['SafariCustom'], customLaunchers});
+    } else {
+      Object.assign(this, {browsers: ['ChromeCustom'], customLaunchers});
```
Will it be easier to read to use a switch statement, or if statements with early returns?

rsimha replied:
Good question. Early returns made the function unnecessarily verbose, and switch statement didn't work because the key in `argv` is different for each case. But we can eliminate all the `Object.assign` repetition (done in last commit). 👍

rileyajones commented on .github/workflows/ie-tests.yml:
```diff
@@ -0,0 +1,22 @@
+      - name: Install Dependencies
+        run: bash ./.github/workflows/install_dependencies.sh
+      - name: Build and Test
+        run: node build-system/pr-check/ie-tests.js
```
It seems odd that you've created this flag in `build-system/pr-check/browser-tests.js` but then still continue to call these files directly. Is there a reason to have both?
```suggestion
        run: node build-system/pr-check/browser-tests.js --browser=edge
```

rsimha replied:
Good observation. The reason this difference exists is that `browser-tests.js` is designed to run on CircleCI in a Docker VM, where it can reuse previously built binaries and run all tests in 3-5 mins. Meanwhile, `ie-tests.js` is a legacy script that runs on GH Actions in a Windows VM, and needs to build the runtime from scratch before running integration tests on IE. It also doesn't run unit / E2E tests.

The plan is to delete `ie-tests.js` later this month once we fully remove all IE 11 code from the repo. This will happen in #35341 (will need to rebase after this PR lands).

samouri commented:
@rsimha FYI: I believe that this PR has introduced a new error for CI on [every commit](https://github.com/ampproject/amphtml/commits/main) to `main`. I can help investigate tomorrow

Example error ([link](https://github.com/ampproject/amphtml/runs/3314949857))
```bash
[18:38:17] ERROR: Error: Build failed with 1 error:
extensions/amp-animation/0.1/parsers/css-expr.js:16:36: error: Could not resolve "../../../../build/parsers/css-expr-impl" 

[18:38:17] 'build' errored after 2m 14s
[18:38:17] Error: Could not compile amp-animation-0.1.max.js
```

rsimha commented:
@samouri That's a known problem that I too was going to look into at some point. Here's more context:
- Until a couple weeks ago, we were only doing minified build + testing on GH actions (win + macos + linux)
- Once testing moved to CircleCI, I added a step to verify builds on all OSs (post-commit only, non-blocking for PRs)
- It was then that I realized we weren't doing unminified builds on win / macos, so I added these lines:

https://github.com/ampproject/amphtml/blob/6e50aadff1e637fe01e3fd7ed8817f7866ca522a/build-system/pr-check/cross-platform-builds.js#L30-L33

The failure you're seeing is from running 31 on Windows. I suspect the bug has to do with hard-coded paths in `css-expr*` that aren't resolving on Windows.

rsimha commented:
Windows unminified build has been fixed. See https://github.com/ampproject/amphtml/pull/35713#issuecomment-901358399.

samouri commented:
Thank you Raghu! You're on 🔥
